sdk/iot/core/src/az_iot_telemetry.c

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_iot_hub_client.h"
+#include <az_contract_internal.h>
+#include <az_result.h>
+#include <az_span.h>
+
+static az_span telemetry_prop_delim = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '?' });
+static az_span telemetry_prop_separator
+    = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '&' });
+static az_span telemetry_null_terminator
+    = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '\0' });
+static az_span telemetry_topic_prefix = AZ_SPAN_LITERAL_FROM_STR("devices/");
+static az_span telemetry_topic_modules_mid = AZ_SPAN_LITERAL_FROM_STR("/modules/");
+static az_span telemetry_topic_suffix = AZ_SPAN_LITERAL_FROM_STR("/messages/events/");
+
+az_result az_iot_hub_client_telemetry_publish_topic_get(
+    az_iot_hub_client const* client,
+    az_iot_hub_client_properties const* properties,
+    az_span mqtt_topic,
+    az_span* out_mqtt_topic)
+{
+  AZ_CONTRACT_ARG_NOT_NULL(client);
+  AZ_CONTRACT_ARG_VALID_SPAN(mqtt_topic);
+  AZ_CONTRACT_ARG_NOT_NULL(out_mqtt_topic);
+
+  az_result result;
+
+  int32_t required_size = 0;
+
+  // Required topic parts
+  required_size += az_span_length(telemetry_topic_prefix)
```

Comment:
>+= [](start = 16, length = 2)

its complicated.

Can you please simplify this? with no +=

required_size = az_span_length(telemetry_topic_prefix)
      + az_span_length(client->_internal.device_id) + az_span_length(telemetry_topic_suffix)
      + az_span_length(telemetry_null_terminator);

Reply:
ah in this case yes